app/src/main/java/gov/hhs/cdc/trustedintermediary/wrappers/Formatter.java

```diff
@@ -0,0 +1,10 @@
+package gov.hhs.cdc.trustedintermediary.wrappers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+public interface Formatter {
+
+    <T> T convertToObject(String input, T clazz) throws JsonProcessingException;
+
+    String convertToString(Object obj) throws JsonProcessingException;
```

Comment:
```suggestion
    <T> T convertToObject(String input, Class<T> clazz);

    String convertToString(Object obj);
```
We shouldn't have anything specific to JSON in this generic interface.  The implementation behind this interface could be parsing protobuf and it wouldn't make any sense to return a `JsonProcessingException`.  So, if you want to throw an exception indicating there was some problem with parsing, then we should throw an exception that is more generic.  Maybe call it `FormatterProcessingException`.  Then the implementation, `JacksonFormatter`, will convert the `JsonProcessingException` to the `FormatterProcessingException` (or perhaps just wrap this exception).

Reply:
Yes, I agree. Makes it more generic and we can use it with many formatters.